### PR TITLE
UA25-015 Added Experimental Client Capabilities and <DOES_NOT_HAVE> pattern

### PR DESCRIPTION
**Pull request by joaopsazevedo**
Added check to confirm that client supports advanced refactorings before sending advanced refactoring code actions.
Added test case and adjusted existing ones by adding a <DOES_NOT_HAVE> pattern. This pattern is used to confirm that the server does not send advanced refactoring code actions to clients that don't support it.

### CODE REVIEW

File: source/tester/tester-tests.adb [hunk 1]
```diff
@@ -63,6 +63,11 @@ package body Tester.Tests is
    function Is_Has_Pattern (List : GNATCOLL.JSON.JSON_Array) return Boolean;
    --  Check if List in form of ["<HAS>", item1, item2, ...]
 
+   function Is_Does_Not_Have_Pattern
+     (List : GNATCOLL.JSON.JSON_Array)
+      return Boolean;
+   --  Check if List in form of ["<DOES_NOT_HAVE>", item1, item2, ...]
```
Comment (reznikmm) [inline source/tester/tester-tests.adb:69]: Could you add the description of "<DOES_NOT_HAVE>" to testsuite/ada_lsp/README.md?

**Review by reznikmm — CHANGES_REQUESTED** [review]

**Comment by reznikmm** [comment]
I guess, the "Added <DOES_NOT_HAVE> pattern" commit (and perhaps part of second commit) could go to the `master`, right?

File: source/tester/tester-tests.adb (hunk 1 above)
Comment (joaopsazevedo) [reply source/tester/tester-tests.adb:69]: Agree.
Added: https://github.com/AdaCore/ada_language_server/pull/847/commits/8db744a58702099d506a192cc8ddd6eea289ed3a#diff-137ea228c3dbbb82abc5fd889e19bf873dac27ffff4380adc10b11a004dd9754R95

**Comment by joaopsazevedo** [comment]
> I guess, the "Added <DOES_NOT_HAVE> pattern" commit (and perhaps part of second commit) could go to the `master`, right?

Agree. Both can go to master. I changed the PR's base to master.

**Review by reznikmm — APPROVED** [review]

**Pull request merged**